src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const arrayStates = [];
+
+  for (const key of actions) {
+    switch (key.type) {
+      case 'addProperties':
+        Object.assign(newState, key.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const action of key.keysToRemove) {
+          delete newState[action];
+        }
+        break;
+
+      case 'clear':
+        newState = {};
+        break;
+    }
```

Comment:
A `default` case should be added to the `switch` statement to handle any unexpected `type` values. This can help with debugging and ensure that the function behaves predictably even with incorrect input.